src/buffer/out/cursor.cpp

```diff
@@ -204,50 +204,50 @@ void Cursor::SetPosition(const COORD cPosition)
     ResetDelayEOLWrap();
 }
 
-void Cursor::SetXPosition(const int NewX)
+void Cursor::SetXPosition(const int NewX) noexcept
 {
     _RedrawCursor();
-    _cPosition.X = (SHORT)NewX;
+    _cPosition.X = gsl::narrow<SHORT>(NewX);

```

Comment:
this is marked noexcept but i thought narrow was the throwing one

Reply:
Urgh, it's actually weird and I should probably undo the one I just did earlier. `gsl::narrow` would have been detected by the static analysis as throwing and it would have errored out if it was a problem.

The trick is that I believe the implementation of `narrow` causes a Fail Fast and isn't actually throwing anything. Since that's not going to come back or in any way continue, it isn't technically violating the `noexcept` marking...